internal/security/secretstore/init.go

```diff
@@ -360,14 +361,26 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, _ *sync.WaitGroup, _ s
 	// Redis 5.x only supports a single shared password. When Redis 6 is released, this can be updated
 	// to a per service password.
 
-	redis5Password, err := cred.GeneratePassword(ctx)
+	redis5Pair, err := getDBCredential("security-bootstrapper-redis", cred, "redisdb")
 	if err != nil {
-		lc.Error("failed to generate redis5 password")
-		os.Exit(1)
-	}
-	redis5Pair := UserPasswordPair{
-		User:     "redis5",
-		Password: redis5Password,
+		if err != errNotFound {
```

Comment:
this returned error message is actually misleading. we should change the message of it. It is currently used for certificate pair not found or not uploaded.

Reply:
This isn't a message. It is used to determine if the `not found error` was returned from Vault. If not found we generate it. Otherwise use it.